src/components/TextField/TextField.tsx

```
@@ -21,13 +22,15 @@ export const TextField: React.FC<Props> = ({
   placeholder = `Enter ${label}`,
   required = false,
   onChange = () => {},
+  customValidate = () => {},
 }) => {
   // generate a unique id once on component load
   const [id] = useState(() => `${name}-${getRandomDigits()}`);
 
   // To show errors only if the field was touched (onBlur)
   const [touched, setTouched] = useState(false);
   const hasError = touched && required && !value;
+  const hasErrorUrl = touched && customValidate(value);
```

`hasErrorUrl` is computed as `touched && customValidate(value)`. This assumes `customValidate` returns `true` when the value is invalid. If you prefer a validator that returns `true` when valid, invert this expression. Make the expected boolean semantics explicit (either via naming, docs, or by returning an error string instead of a boolean). Without clarifying this, callers (e.g., NewMovie) may implement the validator with the opposite expectation and URL validation will fail silently.